timemachine/fe/free_energy.py

```diff
@@ -1165,6 +1165,17 @@ def compute_dense():
 
     U_kl = compute_sparse(max_delta_states) if max_delta_states is not None else compute_dense()
 
+    # Verify that all of the diagonal energies are finite, else a state has blown up
+    diag_idxs = np.diag_indices_from(U_kl)
+    assert np.all(np.isfinite(U_kl[diag_idxs])), "A state is no longer valid"
```

Comment:
super nit: any reason not to just do
```suggestion
    assert np.all(np.isfinite(np.diagonal(U_kl))), "Got non-finite energy while evaluating the energy of replica"
```
?

Could also make assertion message more descriptive.